src/lib/near-fastauth-wallet.ts

```diff
@@ -152,14 +156,119 @@ const FastAuthWallet: WalletBehaviourFactory<
           account.accountId,
           nearAPI.utils.PublicKey.from(accessKey.public_key),
           transaction.receiverId,
-          accessKey.access_key.nonce + index + 1,
+          new BN(accessKey.access_key.nonce).add(new BN(1)),
           actions,
           nearAPI.utils.serialize.base_decode(block.header.hash)
         );
       })
     );
   };
 
+  const _signAndSendWithRelayer = async ({
+    transactions,
+    callbackUrl,
+  }: {
+    transactions: Optional<Transaction, 'signerId'>[];
+    callbackUrl?: string;
+  }): Promise<void> => {
+    const account = _state.wallet.account();
+    const needsFAK = await _getNeedFAK(transactions, account);
+
+    if (needsFAK) {
+      const { closeDialog, signedDelegates } = await _state.wallet.requestSign({
+        transactions: await transformTransactions(transactions),
+        callbackUrl,
+        type: 'delegates',
+      });
+
+      closeDialog();
+      await Promise.allSettled(
+        signedDelegates.map(async (signedDelegate) => {
+          await fetch(relayerUrl, {
+            method: 'POST',
+            mode: 'cors',
+            body: JSON.stringify(
+              Array.from(encodeSignedDelegate(signedDelegate))
+            ),
+            headers: new Headers({ 'Content-Type': 'application/json' }),
+          });
+        })
+      );
+    } else {
+      for (const { receiverId, actions } of transactions) {
+        const signedDelegate = await account.signedDelegate({
+          actions: actions.map((action) => createAction(action)),
+          blockHeightTtl: 60,
+          receiverId,
+        });
+
+        await fetch(relayerUrl, {
+          method: 'POST',
+          mode: 'cors',
+          body: JSON.stringify(
+            Array.from(encodeSignedDelegate(signedDelegate))
+          ),
+          headers: new Headers({ 'Content-Type': 'application/json' }),
+        });
+
+        // Remove the cached access key, otherwise the same nonce will be used for the following transaction
+        delete account.accessKeyByPublicKeyCache[
+          signedDelegate.delegateAction.publicKey.toString()
+        ];
+      }
+    }
+  };
+
+  const _signAndSend = async ({
+    transactions,
+    callbackUrl,
+  }: {
+    transactions: Optional<Transaction, 'signerId'>[];
+    callbackUrl?: string;
+  }): Promise<void> => {
+    const account = _state.wallet.account();
+    const connectedNearAccountWallet = _state.wallet._connectedAccount;
```

Comment:
These two are the same yes? Why not just use account?

Reply:
Idd, they are the same. Updated to use only one